src/com/com_nrf24l01_reg.hpp

```diff
@@ -0,0 +1,283 @@
+}  // namespace instruction_word
+
+namespace rf_config {
+static constexpr std::uint8_t rf_channel = 20;
+}
+
+namespace reg {
+/// configuration register
+namespace config {
+static constexpr std::uint8_t REG_ADDR = 0x0;
+static constexpr std::uint8_t MASK_RX_DR = 6;
+static constexpr std::uint8_t MASK_TX_DS = 5;
+static constexpr std::uint8_t MASK_MAX_RT = 4;
+static constexpr std::uint8_t EN_CRC = 3;
+static constexpr std::uint8_t CRCO = 2;
+static constexpr std::uint8_t PWR_UP = 1;
+static constexpr std::uint8_t PRIM_RX = 0;
+}  // namespace config
+
+/// enable enhanced shock burst auto acknowledgement
+namespace en_aa {
+static constexpr std::uint8_t REG_ADDR = 0x1;
+static constexpr std::uint8_t ENAA_P5 = 5;
+static constexpr std::uint8_t ENAA_P4 = 4;
+static constexpr std::uint8_t ENAA_P3 = 3;
+static constexpr std::uint8_t ENAA_P2 = 2;
+static constexpr std::uint8_t ENAA_P1 = 1;
+static constexpr std::uint8_t ENAA_P0 = 0;
+}  // namespace en_aa
+
+/// enable RX data pipes
+namespace en_rxaddr {
+static constexpr std::uint8_t REG_ADDR = 0x2;
+static constexpr std::uint8_t ERX_P5 = 5;
+static constexpr std::uint8_t ERX_P4 = 4;
+static constexpr std::uint8_t ERX_P3 = 3;
+static constexpr std::uint8_t ERX_P2 = 2;
+static constexpr std::uint8_t ERX_P1 = 1;
+static constexpr std::uint8_t ERX_P0 = 1;
+}  // namespace en_rxaddr
+
+/// setup of address widths
+namespace setup_aw {
+static constexpr std::uint8_t REG_ADDR = 0x3;
+}  // namespace setup_aw
+
+/// setup of automatic retransmission
+namespace setup_retr {
+static constexpr std::uint8_t REG_ADDR = 0x4;
+}  // namespace setup_retr
+
+/// RF channel setting
+namespace rf_ch {
+static constexpr std::uint8_t REG_ADDR = 0x5;
+}  // namespace rf_ch
+
+/// RF setup register
+namespace rf_setup {
+static constexpr std::uint8_t REG_ADDR = 0x6;
+static constexpr std::uint8_t PLL_LOCK = 4;
+static constexpr std::uint8_t RF_DR = 3;
+static constexpr std::uint8_t RF_PWR = 1;
+static constexpr std::uint8_t LNA_HCURR = 0;
+}  // namespace rf_setup
+
+/// Status Register
+namespace status {
+static constexpr std::uint8_t REG_ADDR = 0x7;
+static constexpr std::uint8_t RX_DR = 6;
+static constexpr std::uint8_t TX_DS = 5;
+static constexpr std::uint8_t MAX_RT = 4;
+static constexpr std::uint8_t RX_P_NO = 1;
+static constexpr std::uint8_t RX_P_NO_0 = 0b0;
+static constexpr std::uint8_t RX_P_NO_1 = 0b1;
+static constexpr std::uint8_t RX_P_NO_2 = 0b10;
+static constexpr std::uint8_t RX_P_NO_3 = 0b11;
+static constexpr std::uint8_t RX_P_NO_4 = 0b100;
+static constexpr std::uint8_t RX_P_NO_5 = 0b101;
+static constexpr std::uint8_t RX_FIFO_EMPTY = 0b111;
+static constexpr std::uint8_t TX_FULL = 0;
+}  // namespace status
+
+/// Transmit observe register
+namespace observe_tx {
+static constexpr std::uint8_t REG_ADDR = 0x8;
+static constexpr std::uint8_t PLOS_CNT = 4;
+static constexpr std::uint8_t ARC_CNT = 0;
+}  // namespace observe_tx
+
+/// Carrier detect
+namespace cd {
```

Comment:
This is dangerously short. Even carrier_detect would be short enough

Reply:
Naming taken from nrf24l01 documentation, but changed to rpd as mentioned in the updated docs.